### ISSUE TITLE
💬 [Support Request] Trivial question: item_instance guid overflow, the final death of a server?

### ISSUE THREAD

**Issue by AlterEvo** [cmangos/issues#2169]
## 💬 Support Request

Hello everyone!

I've followed the developing of mangos based cores for years and I've also collaborated to some private projects during these years. Based on wiki, mostly all tables of the char database are using a GUID INT, That, as stated here: http://www.bajb.net/2008/10/mysql-int1-or-int10/, can contain 4294967295 elements. I always asked to my self: Is it possible for such tables having an overflow of data?

There are some discussions on several forums talking about creature guid overflow: https://www.getmangos.eu/forums/topic/1895-efficient-guid-generation/?tab=comments#comment-51131

https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-emulator-servers/wow-emu-questions-requests/431702-creature-guid-overflow.html

However, I'm more afraid about the item_instance table instead. Because if you think about it, a single player could potentially create hundreds of thousands of item GUIDs inside that table in a single year of activity, and I'm not talking about the items that they own, but those ones that a player loots and then sells or destroys (creating huge GUID gaps inside the table itself).

### Version & Environment

ALL

### Steps to reproduce

A realistic case for medium-big international servers: Suppose that you've 2000 players online that are farming, they are looting and destroying 5 items each minute. So in a year:

2000 (average online players) x 5 (average items destroyed per player per minute) x 525600 (minutes in a year) = it's 5 billion ! OVERFLOW

You can play a bit with the numbers but it's still a realistic situation, so, 4 billions are not enough and your server with 2000 players is going to die after 1-2 years? Isn't it?

Am I missing something?

**Comment by Warlockbugs** [comment]
This is a very good question. Client expects a 32bit integer, there is no way around it.

One of the reasons why weekly server maintenance is present on retail is to defrag storage with guids. Along with remapping tables, all logs and backups have to remapped in the same way, so this is a lengthy process.

There is no such solution in CMaNGOS (yet). It should be stated, that this project exists as an academic project first and foremost, hosting a live server is against disclaimer. So specific problems directly tied to this goal, are unexplored for the most part at the moment.

**Comment by AlterEvo** [comment]
Thank you @Warlockbugs ! do you have any source/link where they talk about the "guid storage defrag"?

**Comment by Warlockbugs** [comment]
I don't believe they disclosed any details like that. This is an educated guess deduced from the limitations of the framework. To prove/disprove that, we need to compare internal item guids before and after maintenance on a recently created character and see if they change.

**Comment by jimmybrancaccio** [comment]
I've added the 'Info: Needs Testing' label to this one.
